File: Support/Templates/vb/Files/CommandObject.vb
<Serializable()> _
Public Class CommandObject
  Inherits CommandBase(Of CommandObject)

  Public Shared Function CanExecuteCommand() As Boolean

    'TODO: customize to check user role
    'Return Csla.ApplicationContext.User.IsInRole("")
    Return True

  End Function

  Public Shared Function Execute() As Boolean
    If Not CanExecuteCommand() Then
      Throw New System.Security.SecurityException("Not authorized to execute command")
    End If
    Dim cmd As New CommandObject()
    cmd.BeforeServer()
    cmd = DataPortal.Execute(Of CommandObject)(cmd)
    cmd.AfterServer()
    Return cmd.Result
  End Function

#Region " Client-side Code "

  Public Shared ReadOnly ResultProperty As PropertyInfo(Of Boolean) = RegisterProperty(Of Boolean)(NameOf(Result))
  Public Property Result() As Boolean
    Get
      Return ReadProperty(ResultProperty)
    End Get
    Private Set
      LoadProperty(ResultProperty, value)
    End Set
  End Property


  Private Sub BeforeServer()
    ' implement code to run on client
    ' before server is called
  End Sub

  Private Sub AfterServer()
    ' implement code to run on client
    ' after server is called
  End Sub

#End Region

#Region " Server-side Code "

  <Execute>
  Private Sub ExecuteCommand()

    ' TODO: implement code to run on server
    ' here - and set result value(s)
    Result = True

  End Sub

#End Region

End Class

File: Support/Templates/cs/Files/CommandObject.cs
using System;
using Csla;

namespace Templates
{
  [Serializable]
  public class CommandObject : CommandBase<CommandObject>
  {
    public static bool CanExecuteCommand()
    {
      // TODO: customize to check user role
      //return Csla.ApplicationContext.User.IsInRole("Role");
      return true;
    }

    public static bool Execute()
    {
      if (!CanExecuteCommand())
        throw new System.Security.SecurityException("Not authorized to execute command");

      CommandObject cmd = new CommandObject();
      cmd.BeforeServer();
      cmd = DataPortal.Execute<CommandObject>(cmd);
      cmd.AfterServer();
      return cmd.Result;
    }

    #region Client-side Code

    public static readonly PropertyInfo<bool> ResultProperty = RegisterProperty<bool>(nameof(Result));
    public bool Result
    {
      get { return ReadProperty(ResultProperty); }
      private set { LoadProperty(ResultProperty, value); }
    }

    private void BeforeServer()
    {
      // TODO: implement code to run on client
      // before server is called
    }

    private void AfterServer()
    {
      // TODO: implement code to run on client
      // after server is called
    }

    #endregion

    #region Server-side Code

    [Execute]
    private void ExecuteCommand()
    {
      // TODO: implement code to run on server
      // and set result value(s)
      Result = true;
    }

    #endregion
  }
}
